Realm/RLMManagedArray.mm

```diff
@@ -353,34 +383,63 @@ - (id)valueForKey:(NSString *)key {
-    RLMCollectionSetValueForKey(self, key, value);
+    if ([key isEqualToString:@"self"]) {
+        RLMArrayValidateMatchingObjectType(self, value);
+        RLMAccessorContext context(_realm, *_objectInfo);
+        translateErrors([&] {
+            for (size_t i = 0, count = _backingList.size(); i < count; ++i) {
+                _backingList.set(context, i, value);
+            }
+        });
+        return;
+    }
+    else if (_type == RLMPropertyTypeObject) {
+        RLMArrayValidateMatchingObjectType(self, value);
+        translateErrors([&] { _backingList.verify_in_transaction(); });
+        RLMCollectionSetValueForKey(self, key, value);
+    }
+    else {
+        [self setValue:value forUndefinedKey:key];
+    }
+}
+
+- (size_t)columnForProperty:(NSString *)propertyName {
+    if (_backingList.get_type() == realm::PropertyType::Object) {
+        return _objectInfo->tableColumn(propertyName);
+    }
+    if (![propertyName isEqualToString:@"self"]) {
+        @throw RLMException(@"Arrays of '%@' can only be aggregated on \"self\"", RLMTypeToString(_type));
+    }
+    return 0;
 }
 
 - (id)minOfProperty:(NSString *)property {
-    size_t column = _objectInfo->tableColumn(property);
-    auto value = translateErrors([&] { return _backingList.min(column); }, @"minOfProperty");
+    size_t column = [self columnForProperty:property];
+    auto value = translateErrors(self, [&] { return _backingList.min(column); }, @"minOfProperty");
     return value ? RLMMixedToObjc(*value) : nil;
 }
 
 - (id)maxOfProperty:(NSString *)property {
-    size_t column = _objectInfo->tableColumn(property);
-    auto value = translateErrors([&] { return _backingList.max(column); }, @"maxOfProperty");
+    size_t column = [self columnForProperty:property];
+    auto value = translateErrors(self, [&] { return _backingList.max(column); }, @"maxOfProperty");
     return value ? RLMMixedToObjc(*value) : nil;
 }
 
 - (id)sumOfProperty:(NSString *)property {
-    size_t column = _objectInfo->tableColumn(property);
-    return RLMMixedToObjc(translateErrors([&] { return _backingList.sum(column); }, @"sumOfProperty"));
+    size_t column = [self columnForProperty:property];
+    return RLMMixedToObjc(translateErrors(self, [&] { return _backingList.sum(column); }, @"sumOfProperty"));
 }
 
 - (id)averageOfProperty:(NSString *)property {
-    size_t column = _objectInfo->tableColumn(property);
-    auto value = translateErrors([&] { return _backingList.average(column); }, @"averageOfProperty");
+    size_t column = [self columnForProperty:property];
+    auto value = translateErrors(self, [&] { return _backingList.average(column); }, @"averageOfProperty");
     return value ? @(*value) : nil;
 }
 
 - (void)deleteObjectsFromRealm {
+    if (_type != RLMPropertyTypeObject) {
+        @throw RLMException(@"Cannot delete objects from RLMArray<%@>: only RLMObjects can be deleted.", RLMTypeToString(_type));
```

Comment:
Clearing the array seems close enough in terms of semantics to what happens with arrays containing objects that it may be simpler to do that rather than throwing here?

Reply:
The public API that this is called by (`-[RLMRealm deleteObjects:]`) can't support deleting collections of non-objects (what would `[realm deleteObjects:@[@0]]` do?).